### ISSUE TITLE
"invalid typeclass name" error when using with Razzle

### ISSUE THREAD

**Issue by pranksinatra** [microstates/microstates.js#342]
I've been working deploying a [Razzle](https://github.com/jaredpalmer/razzle) app that includes microstates as a dependency, and have been getting this error:

`Uncaught Error: invalid typeclass name`

To isolate it, I've deployed a starter Razzle app with microstates as the sole dependency, and can confirm that without microstates everything works fine, but by adding it, server-rendering works but the client aborts with the above error message. You can see the deployed app here:
https://with-now-v2-lkd7cvw71.now.sh/

Code in App.js (nothing special, just rendering a microstates Number type)
```js
import './App.css';
import React from 'react';
import { create, valueOf } from 'microstates';

let meaningOfLifeAndEverything = create(Number, 42);

const App = () => (
  <div>
    Welcome to Razzle. Meaning of life = {meaningOfLifeAndEverything.state}
  </div>
);

export default App;
```

I'm happy to move all the code into a repository if that'd be helpful for debugging. My best guess is that this has to do with the Razzle build process (babel, terser, etc). I did some reading and see that you fixed a similar issue with funcadelic class names being removed during uglification a few months ago.

If you have suggestions for things to try, please let me know. Thanks!

**Comment by cowboyd** [comment]
@pranksinatra Thank you for taking the time to let us know about this problem. You're absolutely right: it's one that we've seen before https://github.com/cowboyd/funcadelic.js/issues/63 Although I _thought_ we'd quashed it :)

This happened before because uglify was removing the implicit `name` property that was added to constructors. Causing the following exception to be triggered. https://github.com/cowboyd/funcadelic.js/blob/a3b8aea1353329369816de697115bbda1ca23e88/src/typeclasses.js#L13-L17

We mitigated this by making every class, get its own name _explicitly_. Like this one here https://github.com/microstates/microstates.js/blob/master/src/tree.js#L6

I think the following would help diagnose the issue:

1. microstates version
2. complete stacktrace

**Comment by cowboyd** [comment]
The workaround when using uglify was to set the `keep_classnames` option https://github.com/microstates/ember/pull/98#issue-238878513 

Perhaps there is an equivalent with Terser?

**Comment by taras** [comment]
You can see in the built output that the class names are missing.

<img width="588" alt="image" src="https://user-images.githubusercontent.com/74687/55090491-9314d800-5085-11e9-9439-d22ed975dccc.png">

If Razzer used `keep-classnames` flag then it would work.

I believe the change needs to be applied here https://github.com/jaredpalmer/razzle/blob/8fed32b57ec5b75c421e04f00a4081da687ab6d6/packages/razzle/config/createConfig.js#L516

> We mitigated this by making every class, get its own name explicitly.

We did this in Microstates but not in Funcadelic. Should we apply the same fix to Funcadelic?

**Comment by taras** [comment]
@pranksinatra I was able to reproduce the issue with Razzle and test a fix. Please standby for a fix.

**Comment by taras** [comment]
@pranksinatra I just released 0.14.0 which includes a fix for your issue and tested it with the Razzle app. I'm going to close this issue but feel free to re-open it if it's not resolved. 

*Note*: please make sure to bump the version of microstates and run build before trying it.
